src/components/transport_manager/src/usb/libusb/usb_connection.cc

```diff
@@ -192,43 +194,50 @@ bool UsbConnection::PostOutTransfer() {
   if (LIBUSB_SUCCESS != libusb_ret) {
     LOG4CXX_ERROR(
         logger_,
-        "libusb_submit_transfer failed: " << libusb_error_name(libusb_ret)
-                                          << ". Abort connection.");
-    AbortConnection();
+        "libusb_submit_transfer failed: " << libusb_error_name(libusb_ret));
     LOG4CXX_TRACE(logger_,
                   "exit with FALSE. Condition: "
                       << "LIBUSB_SUCCESS != libusb_fill_bulk_transfer");
-    return false;
+    return TransportAdapter::FAIL;
   }
   LOG4CXX_TRACE(logger_, "exit with TRUE");
-  return true;
+  return TransportAdapter::OK;
 }
 
 void UsbConnection::OnOutTransfer(libusb_transfer* transfer) {
   LOG4CXX_TRACE(logger_, "enter with  Libusb_transfer*: " << transfer);
-  sync_primitives::AutoLock locker(out_messages_mutex_);
-  if (transfer->status == LIBUSB_TRANSFER_COMPLETED) {
-    bytes_sent_ += transfer->actual_length;
-    if (bytes_sent_ == current_out_message_->data_size()) {
-      LOG4CXX_DEBUG(
+  auto error_code = TransportAdapter::OK;
+  {
+    sync_primitives::AutoLock locker(out_messages_mutex_);
+    if (LIBUSB_TRANSFER_COMPLETED == transfer->status) {
+      bytes_sent_ += transfer->actual_length;
+      if (current_out_message_->data_size() == bytes_sent_) {
+        LOG4CXX_DEBUG(
+            logger_,
+            "USB out transfer, data sent: " << current_out_message_.get());
+        controller_->DataSendDone(
+            device_uid_, app_handle_, current_out_message_);
+        error_code = PopOutMessage();
+      }
+    } else {
+      LOG4CXX_ERROR(
           logger_,
-          "USB out transfer, data sent: " << current_out_message_.get());
-      controller_->DataSendDone(device_uid_, app_handle_, current_out_message_);
-      PopOutMessage();
+          "USB out transfer failed: " << libusb_error_name(transfer->status));
+      controller_->DataSendFailed(
+          device_uid_, app_handle_, current_out_message_, DataSendError());
+      error_code = PopOutMessage();
+    }
+    if (current_out_message_.use_count() == 0) {
+      libusb_free_transfer(transfer);
+      out_transfer_ = NULL;
```

Comment:
@SNiukalov `nullptr`

Reply:
@AByzhynar 
Thanks, fixed in:https://github.com/smartdevicelink/sdl_core/pull/3332/commits/581a1b9d0e927f22754b6baa59c6df89ec19c721